packages/data-point/lib/reducer-types/resolve.js

```diff
@@ -15,28 +16,55 @@ const reducers = Object.assign({}, ReducerHelpers, {
   [ReducerPath.type]: ReducerPath
 })
 
+/**
+ * @param {Reducer} reducer
+ * @return {boolean}
+ */
+function hasDefault (reducer) {
+  return !!reducer[DEFAULT_VALUE]
+}
+
+module.exports.hasDefault = hasDefault
+
+/**
+ * @param {Reducer} reducer
+ * @throws if reducer is not valid
+ * @return {Function}
+ */
+function getResolveFunction (reducer) {
+  const reducerType = reducers[reducer.type]
+  if (reducerType) {
+    return reducerType.resolve
+  }
+
+  throw new Error(`Reducer type '${reducer.type}' was not recognized`)
+}
+
 /**
  * apply a Reducer to an accumulator
  * @param {Object} manager
  * @param {Accumulator} accumulator
  * @param {Reducer} reducer
  * @returns {Promise<Accumulator>}
  */
-function resolve (manager, accumulator, reducer) {
+function resolveReducer (manager, accumulator, reducer) {
   // this conditional is here because BaseEntity#resolve
   // does not check that lifecycle methods are defined
   // before trying to resolve them
   if (!reducer) {
     return Promise.resolve(accumulator)
   }
 
-  const reducerType = reducers[reducer.type]
-  if (!reducerType) {
-    throw new Error(`Reducer type '${reducer.type}' was not recognized`)
+  let resolve = getResolveFunction(reducer)
+  // NOTE: recursive call
+  resolve = resolve(manager, resolveReducer, accumulator, reducer)
+  if (hasDefault(reducer)) {
+    const _default = reducer[DEFAULT_VALUE].value
+    const afterResolve = reducers.ReducerDefault.resolve
+    return resolve.then(acc => afterResolve(acc, _default))
```

Comment:
I thought about using the default when a reducer throws an error:

```js
return resolve.catch(e => ({ value: null })).then(acc => afterResolve(acc, _default))
```

Any thoughts on doing this? It could even be a setting that's passed to `withDefault`:

```js
const throwError = () => { throw new Error() }

const reducer = withDefault(throwError, 'DEFAULT_VALUE', { catch: true })

dataPoint.resolve(reducer, {}) // => 'DEFAULT_VALUE'
```